From 60ee5191a0c074251862a15b12afdc9db0b2df38 Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Thu, 17 Nov 2016 15:36:54 -0300
Subject: [PATCH] configure.ac: invert order for wayland-scanner check

When cross-compiling the .pc file might point to the wrong
wayland-scanner binary (target rather than host) resulting in a
non-executable and wrong scanner.
Try searching the PATH first, and if that fails fall back into
pkg-config.

[Vincent: tweak patch for 17.1.1 version]

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
---
 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2c7e636..0198d52 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2174,11 +2174,11 @@ if test "x$with_platforms" != xauto; then
     with_egl_platforms=$with_platforms
 fi
 
-PKG_CHECK_MODULES([WAYLAND_SCANNER], [wayland-scanner],
-        WAYLAND_SCANNER=`$PKG_CONFIG --variable=wayland_scanner wayland-scanner`,
-        WAYLAND_SCANNER='')
+AC_PATH_PROG([WAYLAND_SCANNER], [wayland-scanner])
 if test "x$WAYLAND_SCANNER" = x; then
-    AC_PATH_PROG([WAYLAND_SCANNER], [wayland-scanner], [:])
+	PKG_CHECK_MODULES([WAYLAND_SCANNER], [wayland-scanner],
+		WAYLAND_SCANNER=`$PKG_CONFIG --variable=wayland_scanner wayland-scanner`,
+		WAYLAND_SCANNER='')
 fi
 
 # Do per-EGL platform setups and checks
-- 
2.7.3

